telemetry/packages/ui/app/components/views/state-machine/edges.ts

```diff
@@ -1,94 +1,218 @@
+    },
     markerEnd: arrow,
   },
   {
-    id: 'nominal-braking-stopped',
-    source: 'nominal-braking',
-    target: 'stopped',
-    type: 'step',
+    id: 'accelerate-lim-brake',
+    source: 'accelerate',
+    target: 'lim-brake',
+    sourceHandle: 'bottom-bottom',
+    targetHandle: 'top-top',
+    type: 'smoothstep',
+    pathOptions: {
+      borderRadius: 20,
+    },
     markerEnd: arrow,
   },
   {
-    id: 'nominal-braking-failure-braking',
-    source: 'nominal-braking',
-    target: 'failure-braking',
-    sourceHandle: 'top',
-    type: 'step',
+    id: 'lim-brake-friction-brake',
+    source: 'lim-brake',
+    target: 'friction-brake',
+    sourceHandle: 'bottom-bottom',
+    targetHandle: 'top-top',
+    type: 'smoothstep',
+    pathOptions: {
+      borderRadius: 20,
+    },
     markerEnd: arrow,
   },
   {
-    id: 'stopped-off',
+    id: 'friction-brake-stop-levitation',
+    source: 'friction-brake',
+    target: 'stop-levitation',
+    sourceHandle: 'right',
+    targetHandle: 'left',
+    type: 'smoothstep',
+    pathOptions: {
+      borderRadius: 20,
+    },
+    markerEnd: arrow,
+  },
+  //
+  {
+    id: 'stop-levitation-stopped',
+    source: 'stop-levitation',
+    target: 'stopped',
+    sourceHandle: 'bottom',
+    targetHandle: 'top',
+    type: 'smoothstep',
+    pathOptions: {
+      borderRadius: 20,
+    },
+    markerEnd: arrow,
+  },
+  {
+    id: 'stopped-battery-recharge',
     source: 'stopped',
-    target: 'off',
-    type: 'step',
+    target: 'battery-recharge',
+    sourceHandle: 'bottom',
+    targetHandle: 'top',
+    type: 'smoothstep',
+    pathOptions: {
+      borderRadius: 20,
+    },
     markerEnd: arrow,
   },
   {
-    id: 'failure-braking-failure-stopped',
-    source: 'failure-braking',
-    target: 'failure-stopped',
-    type: 'step',
+    id: 'battery-recharge-capacitor-discharge',
+    source: 'battery-recharge',
+    target: 'capacitor-discharge',
+    sourceHandle: 'bottom',
+    targetHandle: 'top',
+    type: 'smoothstep',
+    pathOptions: {
+      borderRadius: 20,
+    },
     markerEnd: arrow,
   },
   {
-    id: 'failure-stopped-off',
-    source: 'failure-stopped',
-    target: 'off',
+    id: 'capacitor-discharge-safe',
+    source: 'capacitor-discharge',
+    target: 'safe',
+    sourceHandle: 'bottom',
     targetHandle: 'top',
-    type: 'step',
+    type: 'smoothstep',
+    pathOptions: {
+      borderRadius: 20,
+    },
+    markerEnd: arrow,
+  },
+  {
+    id: 'failure-braking-capacitor-discharge',
+    source: 'failure-braking',
+    target: 'capacitor-discharge',
+    sourceHandle: 'right',
+    targetHandle: 'left',
+    type: 'smoothstep',
+    pathOptions: {
+      borderRadius: 20,
+    },
     markerEnd: arrow,
   },
 ];
+
+// Generates edge definitions for a given mode of operation
+export const writeEdges = (mode: ModeType): CustomEdgeType[] => {
+  return edges
+    .filter((edge) => {
+      return !MODE_INACTIVE_STATES[mode].includes(
```

Comment:
This check is used a few times, might be worth creating a function to do this check so it's a bit cleaner. E.g. could have `getActiveStates(mode)` which returns all the states that aren't in `MODE_INACTIVE_STATES`.